File: Web/produit.php
<?php
header('Content-Type: application/json');
$bdd = new PDO('mysql:host=172.19.0.13;dbname=tpjson', 'tpjson', 'tpjson');

if(isset($_GET['id'])) {
    $sql = $bdd->prepare("SELECT `reference`, `libelle`, `prixUnitaire` FROM `Produit` WHERE reference = :numero");
    $sql->bindParam(':numero', $_GET['id'], PDO::PARAM_STR);
    $sql->execute();
    $produit = $sql->fetch(PDO::FETCH_ASSOC);
    echo json_encode($produit);
} else {
    $sql = $bdd->prepare("SELECT * FROM Produit");
    $sql->execute();
    $produit = $sql->fetchAll(PDO::FETCH_ASSOC);
    echo json_encode($produit);
}
?>

File: TPJson/Produit.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TPJson
{
    class Produit
    {
        public string reference;
        public string libelle;
        public float prixUnitaire;

        public string Reference { get => reference; set => reference = value; }
        public string Libelle { get => libelle; set => libelle = value; }
        public float PrixUnitaire { get => prixUnitaire; set => prixUnitaire = value; }

        public Produit(string reference, string libelle, float prixUnitaire)
        {
            this.Reference = reference;
            this.Libelle = libelle;
            this.PrixUnitaire = prixUnitaire;
        }
    }
}
